### ISSUE TITLE
Implement sound design

### ISSUE THREAD

**Issue by jbphet** [phetsims/ratio-and-proportion#9]
@Ashton-Morris, @emily-phet, and I (@jbphet) did some brainstorming on the sound design for the proportion sim and thought we should write some of these things down.

- Use two sound generators that are producing the same note when in correct proportion, and when the proportions aren't correct they would produce different notes and thus beat frequencies.
- Same idea - two sound generators - but produce the notes periodically so that it doesn't drive people crazy with continuous sounds
- Two rhythmic beats that are distinct and line up when the proportions are correct - in a fun way, using double time, triplets, etc, but get out of sync if the proportions are incorrect.  They'd speed up or slow down the overall tempo as the fundamental sizes change.

**Comment by jbphet** [comment]
I made a quick recording to demonstrate beat frequencies.  In the explanation at the beginning I probably should have said that I'm "playing the same note on two different strings" instead of that I'm "playing the same note".

Here's a link: https://drive.google.com/file/d/1AmrwecgS4_QDzztTGWdeK7pM2TMpIomE/view

**Comment by zepumph** [comment]
In the above commit. I made a refactor that exposed what I call the `proportionFItnessProperty` in `ProportionModel`. This is a value between 0 and 1. When 1 the proportion is "perfect", and 0 means it is outside of the tolerance range. I did this with the hope that it would be easy to monitor for a simple "fitness" sound that can change in the same way as we do for color. Let me know if I can be of more help.

In my opinion having something (even very simple) implemented by next meeting seems very valuable for discussion. Good luck!

**Comment by emily-phet** [comment]
@Ashton-Morris - please propose something (could be related to beat frequencies or anything else) that is reasonably pleasant and can be added to the sim quickly. I'd like to get some sound in asap so there's some auditory feedback available as the team brainstorms sound and other features. Let's start with something that aligns with the color change in the sim, as @zepumph [indicated](https://github.com/phetsims/proportion/issues/9#issuecomment-611166721). 

@jbphet @zepumph Please work together to add the first sound. @jbphet if you're able to begin showing @zepumph the ropes on how to add sound to sims with Tambo, that would be ideal, but if depending on where you're at with your other rocks, you might do this later in the quarter and proceed with whatever is fastest now.

**Comment by Ashton-Morris** [comment]
@emily-phet I have a few ideas ready but I cant seem to properly clone the directory that has Proportion. 

It would be much quicker if someone could do a simple screen recording of a proportion slider slowly going up and down. I can then do a few mockups of how sound would work and can upload it here.

If anyone is willing to do that please let me know.

**Comment by emily-phet** [comment]
@Ashton-Morris - I'll try to do a screen recording for you later today. Here's a link to the latest dev version, in case that's helpful in the meantime.
https://bayes.colorado.edu/dev/html/proportion/1.0.0-dev.5/phet/proportion_en_phet.html

**Comment by Ashton-Morris** [comment]
@emily-phet Don't worry about a screen recording, I can do that. I didn't know there was a dev version. That all I needed.

I'll work on some mockups asap

**Comment by emily-phet** [comment]
@Ashton-Morris I figured that might have been the issue. :) 

When you make the screen recording, keep in mind that in some scenarios, learners will be moving each object one at a time, but in the more desirable scenarios, they're moving both at the same time (e.g., on a touch screen, or with tangible objects). Just thought that might impact your thoughts re sound.

**Comment by Ashton-Morris** [comment]
I worked on some ideas this morning. You should be able to view these in your browser. I will do my best to explain my thinking with each one.  @emily-phet Please assign the proper people for review

- [Proportion_C_Major_Sine.m4v](https://drive.google.com/open?id=1npUOcMoX-bPlTxfD_UpL0jjpUQW8J1WF)
Max out of proportion = Low volume and only a C sine wave tone
Max in proportion = Added E and G notes that fade in which creates a C Major Chord
What you're hearing  = A C Major Chord fading in and out along with proportion

- [Proportion_Fifth_Sine.m4v](https://drive.google.com/open?id=1QvR6EIPUvJxK1a3gx7Pxbh_gVoQRB6ph)
Max out of proportion = Two C Sine Waves and No Volume 
Max in proportion = Two Sine waves C and G creating a perfect fifth
What you're hearing = A kind of wobble/dissonance as they are out of proportion and a cleaning up/harmony sound when they are in.


- [Proportion_One_Semitone_Sine.m4v](https://drive.google.com/open?id=1xn8tbqictHiYfQmK7CD0GHTkhGrqNrcS)
Similar to above
Max out of proportion = Two Sine Waves one C and one B and No Volume 
Max in proportion = Two Sine waves playing the same note C
What you're hearing = A kind of wobble/dissonance as they are out of proportion and a cleaning up/unity of the notes sound when they are in.


- [Proportion-Rhythm.m4v](https://drive.google.com/open?id=18szF48Q5Grw_Ec7NYVGJfMTqfGgQi54L)
Max out of proportion = A single rhythm at a low volume
Max in proportion = That same rhythm being played at 2X and 4X the speed fading in in volume
What you're hearing = A kick drum like sound going by it self then a more complex rhythm rising in volume as Proportion is reached. 



- [Proportion-Vibrato.m4v](https://drive.google.com/open?id=10UHYCfgQq7nSjcF-8UKiIZd02n51XEB9)
Max out of proportion = A single tone with vibrato and a low volume
Max in proportion = A single tone with no vibrato
What you're hearing = A single tone going from wobbly/vibrato to straight and pure and louder as Proportion is reached. 

- [Proportion-Random-Clicks.m4v](https://drive.google.com/open?id=1lZaTqkxFGvtQ_krm--YKzjV_8_TLzH5C)
Max out of proportion = A single repeating thump/click that has it's pitch randomly changed by 16 semitones each time it repeats at a low volume
Max in proportion = A single repeating thump/click playing a consistent rhythm at a louder volume
What you're hearing = A single repeating thump/click that goes from being in disarray (playing random versions of itself) then fading into a consistent rhythm when the sim is in Proportion.

**Comment by emily-phet** [comment]
@Ashton-Morris Thanks for these! 
@zepumph @jbphet Can we have a version with two sound options - with "Proportion-Vibrato" and "Proportion-Random Clicks" (random clicks as the default)? 
If it's not time consuming to do, having a third option of "Proportion_C_Major_Sine" would be nice to have in as well. 

To ensure those playing with at-home tangible versions has some auditory feedback to consider, this would be my top priority effort for Proportion development this week.

**Comment by emily-phet** [comment]
@Ashton-Morris Here are my thoughts:
- C_Minor_Sine - this sounds very pleasant to my ear. My concern is that from the video I wasn't sure if the feedback tied strongly enough to the goal state of "at ratio". That state needs to simultaneously stand out clearly. If this one gets implemented in the first options set, my hunch is that it will feel like it stands out more so than watching the video. Let's see!

- Fifth Sine and One_Semitone_Sine - both of these sounded slightly less pleasant to my ear than C_Minor_Sine, and had the same potential issue of not quite strong enough of a highlight of the "at ratio" state.

- Rhythm - I thought this approach might muddy the waters a bit at cueing the 'at ratio' state. If we feel like pursuing a rhythm-based approach makes sense to move forward with, I think we'll want one with a really strong change when very very close, and right at, ratio.

- Vibrato - I thought this one had the strongest cueing to "tune" the ratio, though slightly less pleasant to my ear than the C_Minor_Sine, and the random clicks.

- Random Clicks - I really liked this one, and I'm curious what others think. It sounds interesting and pleasant, while also providing some strong cueing when you're "at ratio".

**Comment by Ashton-Morris** [comment]
I have a single click and a loop of them below for implementation. 

In my DAW I have the clicks set to randomize between 0 and +16 semitones. The Randomizer's rate (like the rate of an LFO) is set to 30HZ. Then its set to 100% out of proportion = 100% of that randomizer effecting the clicks, and In proportion = 0% that randomizer effecting those clicks. 

[proportion-random-clicks-loop.mp3](https://drive.google.com/open?id=1ImH89xhalYDmUvqfWhMB9bO3fWSRsx2x)
[proportion-random-clicks-single.mp3](https://drive.google.com/open?id=1B9U776ePYyGVdlFU5dPEOPoSG17R2Ot9)



For the Proportion_C_Major_Sine you could play sine's with web audio. That would be: Play C3 when it is 100% our of proportion and then attach the volume of the sine waves E3 and G3 to how in proportion the sim is. Thats how I had it set up.

**Comment by zepumph** [comment]
I, with tons of great help from @jbphet, added modes for the vibrato and random clicks sounds. The random clicks is on by default. The vibrato is currently just two sine waves at 220hz where one can be moved up to 230hz depending proportion fitness.

**Comment by zepumph** [comment]
After the above commits, we successfully have mechamarkers input working with the two sounds discussed during today's meeting. 

Summary of what is currently in the sim:
* A prototype of Vibrato
* A prototype of Random Clicks
* A prototype that just changes pitch based on the fitness
* Sound working for keyboard/drag/marker inputs

Next I will try to add C_Major sine as the final step for the day.

**Comment by zepumph** [comment]
Above I added the CMajor sine wave generator. It sounds really fun. I'm excited to be working on the sound design for this sim. I got perhaps a bit carried away and added a high C (C4) only when the proportion was perfect. This made for a direct cue from sound for when the user got it totally right. I'm not sure that this implementation is the best, but I really like a specific sound for when (and only when) the "fitness" of the proportion is exactly perfect (in the model a value of 1). 

Dev version coming soon.

**Comment by BLFiedler** [comment]
@terracoda noted that the sound plays continuously on keyboard focus of either cursor without directional input. 

It's not clear what the preferred feedback should be right now, but something we should decide at some point.

**Comment by zepumph** [comment]
Yes that was done purposefully. After I tried it both ways (based on focus/blue and based on keydown/keyup), it felt the committed way was much easier to tell if there was success or not because it was much more continuous.

**Comment by BLFiedler** [comment]
A note from an interview using 1.0.0-dev.9:
- With Random Clicks enabled, the user thought they perceived an increase in the speed of the clicks even when they had not met the threshold for the increase in frequency to begin. Looking for signal in the noise type situation.

**Comment by emily-phet** [comment]
@BLFiedler Glad to hear the sound cueing seems helpful for finding the target ratio! 
Please create a spreadsheet where you can summarize your findings for each interview, regarding specific categories - visual display, auditory display, interaction pathway, etc. You'll also want some fields for sim version, device, and some basic demographics. Let's add on to this as we do more interviews. You might reach out to Brianna (she's on Slack) for an example of some spreadsheets she'd made in this ballpark.

Also, can you 1) clip out the proportion section of these interviews, and 2) edit out all screen area that is not the sim (e.g., people's browser tabs, etc.). Sometimes there is identifying info in the top of the browser window. You can use Camtasia to do this really fast. Let me know if you need some help with this. 3) Put all of these de-identified clips in a shared folder, for the team.

**Comment by emily-phet** [comment]
@jbphet @Ashton-Morris 
Just a heads up, we'll be collecting more feedback re sound from the design team this week. So for this week sound is otherwise on hold, unless there are some specific topics you'd like to bring up when we meet on Tuesday.

**Comment by emily-phet** [comment]
A few thoughts on the random click sound - after playing with the sim + mechamarkers for a bit, I think the random click sound is the least helpful, and most challenging to listen to after awhile. I think tuning more when the change in rate happens will help. Thought not sure if it will help enough to put this sound as the frontrunner. Implementation of the grid representation could help with the "tuning" process for this sound, which may happen this week.

**Comment by emily-phet** [comment]
More thoughts - we should probably start a sound design doc to put these.
- I have liked the vibrato sound the most while playing with Mechamarkers. The current sound is good at indicating nearish-vs-not, but once you hone in on "keeping the screen green" while moving the markers up and down, the sounds don't help much. We need the sound to indicate 1) clearly when exactly at ratio, and to 2) change significantly in the range of close-very close -super close.

**Comment by emily-phet** [comment]
@zepumph Can you make "Vibrato" be the default sound (leaving all current options)?

**Comment by zepumph** [comment]
Done above.

**Comment by Ashton-Morris** [comment]
I came up with more examples of how it could sound when you are in proportion but you are also staying in proportion and moving up and down. @emily-phet please assign people as necessary.


- [Proportion_Applause.mp4](https://drive.google.com/open?id=1KdcGTqZImXd5dstDTOrAWYymfNTyrlGq)
- [Proportion_FM_Synthesis.mp4](https://drive.google.com/open?id=1XmCDzPactzgSyZUhcOnugeq-rMZ52zFo)
- [Proportion_Major_Chord.mp4](https://drive.google.com/open?id=1ravPL3WCA39SVmqT9iv_IjM1qNkKzLwB)
- [Proportion_Moving_Sine_Example.mp4](https://drive.google.com/open?id=1-XAECYno04lg4Kq3ZqgptuN1QR6YedM0)
- [Proportion_Stacc_Sine.mp4](https://drive.google.com/open?id=1xCUzyp0yD8ptSKLsXtWO4_UIxUTvWgYo)
- [Proportion_Stacc_Sine_w_reverb.mp4](https://drive.google.com/open?id=16ggt54oCIFCWJet1q8VKihU73nPUyEtn)
- [Proportion_Strings.mp4](https://drive.google.com/open?id=1BQfLsp5lPN2QDFXTYh6juhwXEPOPsLof)

**Comment by emily-phet** [comment]
@Ashton-Morris These are wonderful! I love these staged approaches you're taking. 

I think a different combination of the staged approaches this set of demos may be worthwhile. I believe we want learners to experience this sequence: 
1) something going "out of phase" then "into phase" [corresponding with the sim starting at ratio (good), then going out of ratio as the learner first moves the object, then going back to ratio (good) as learner nears same or new ratio arrangement] They'll hear this multiple times, as the move the markers around exploring the space.
2) going from "in phase" (good) to "better" [corresponding to being at ratio, then moving the markers such that markers stay at ratio]. In your demos above, all had a feeling of transition in them, but not all had a clear feeling of going from "good" to "better". Most had a feel (to me) of going from "good" to "different". 

Proportion_Strings and Proportion_Applause had the most "good" to "better" feel to me. The transition to the strings in Proportion_Strings felt like a bonus, as it's recognizably going from synthesized techo-sound to a recognizable instrument (which can be easily talked about verbally and is also easier on the ears). 

The transition including the applause in Proportion_Applause was also clearly cueing "good" to "better", once it was clear you were hearing applause. When it first starts (and for learners it would likely start/stop/start/stop for a bit until they get the hang of the movement), it's not 100% clear it's applause. But the basic idea of it was very nice.

It would be great to hear the following:
- an updated demo of Proportion_Strings with applause coming in about halfway into the strings. (I'd like to hear if that feels like overkill altogether).
- 1 or 2 more demos like Proportion_Strings, but with other timbres (instruments) instead of strings. Think things that middle schoolers might feel good about getting as feedback. If in a classroom, we'd want them to be saying, "hey, how do I get it to make *that* sound" when they hear a neighbor getting the "better" sound.
- A demo similar to Proportion_Stacc_Sine  that starts with staccato, then becomes strings. 
- A demo similar to Proportion_Stacc_Sine  that starts with staccato, then becomes strings, then layers in applause over the strings.

And, because I know you like a challenge - double bonus if any of these can include a cello sound. Dor (father of this sim idea) plays the cello. :)

**Comment by emily-phet** [comment]
@BLFiedler please add any thoughts you have as well.

**Comment by BLFiedler** [comment]
**Stage 2 (in prop):** My general impression is that the movement toward a neutral, single tone (from something else) is very successful in generating that feeling of "I found it!". This sound can possibly be more pleasant to listen to for a long period of time, but I don't think it HAS to be. What we need it to be is pleasant enough, but not as nice and rewarding as the next tone. This leads me to...

**Stage 3 (maintain prop):** I think Emily mentioned this, but overall this sound should be BETTER (oh my, subjective), than the in proportion sound. This may be richer or more pleasant or runner, but also should not at all sound like the sounds we heard before (out of proportion or searching sounds). Some of the examples re-introduce some reverb/vibrato (I apologize for confounding those two), which to me may cue the learner to believe they've left the Success they have already achieved, when in fact they've reached an even higher success. I like the Strings for that. I like the idea of the applause for the same reason, but personally find it a little jarring.

_I'm going to distinguish between two phases in the Out of Proportion (Stage 1)_

**Stage 1A (way out of prop):** I think both the kind of reverb-y vibrato, as well as the staccato frequency, are successful in generating that sense of "I'm searching for something and it's guiding me toward it". Personally, I find the staccato effect to give _stronger feedback for distance from proportion and the sense that I am looking for something._ 

**Stage 1B (nearing prop):** 
1. The dissonant sine waves "slow down" and resolve into the success tone, which I think is smooth and "feels good". However, it's a little muddy when you're near the proportion but not quite on it. 
2. The staccato frequency becomes very rapid and the success tone fades in. Without playing sim version with this implemented, I get the sense that this is a better contrast from out of proportion to in proportion. I would actually like to hear a version where there is a more rapid fade-in of the success tone (though we may need to consider tolerance and the visual fade-in as well?)

**Comment by Ashton-Morris** [comment]
@zepumph I have some files for you [which are all stored here](https://drive.google.com/open?id=10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg). Some of these are .wav so we can have better loops during the prototype phase.

I added a staccato sine but realistically that's something we'd need to do in web audio programmatically. 

applause-loop.wav
staccato-sine.mp3
strings-loop-c5.wav
cello-loop-c4.wav
synth-strings-loop-c4.wav
bowed-piano-loop-c4.wav
cello-tremolo-loop-c4.wav

**Comment by jbphet** [comment]
I listened to the video demos after today's sound design meeting and agree with what was essentially decided in the meeting, which is that we should prototype the "Proportion_Strings" version and try adjusting some of the tolerance parameters to see if we can get it to work in a way that is viable.

**Comment by jbphet** [comment]
I'm going to unassign myself for now, since I don't think there is anything specific that I need to contribute at this point.  @zepumph - I'm happy to help with the implementation if I'm needed, just assign it back to me if so and/or we can set up a time to collaborate on it.

**Comment by zepumph** [comment]
After lots of design discussion today, with @Ashton-Morris playing many possibilities for us, we have decided on the following for the next steps:

* Add the chorus option for the velocity sound, (currently strings)
* Add a new ratio sound of marimba tones, increasing in frequency (how often it is played) the closer to successful ratio it gets. Then at a threshold it turns into a root/third tremolo loop. I will take the marimba sound from `Tambo`. (use `bright-marimba.mp3`, the "short" is available if they are bleeding into each other). 
* We will also add an option, same algorithm as above, but with pizz strings instead of marimba.
* We want to "play with silence," and to remove the continuous sound when the ratio "fitness" is equal to 0. This means we only get sound when you are "getting close" to the ratio. Tagging @emily-phet to see if she disagrees. 
* We will hold off on fading the ratio sound out to bring the choir sound in. This fine tuning can be done later, once sounds are more solidified.
* @Ashton-Morris will send the Choir sound (to be looped) as an mp3 and wav. I will try using the mp3 first to see if it works, otherwise we will use the `wav` and take the size hit.

**Comment by zepumph** [comment]
Marimba staccato plus tremolo to third is in place on master and the default sound. I also have turned off sound when the fitness is at 0. Next steps are to add in choir and pizz sound options once @Ashton-Morris has them ready. Over to him.

**Comment by emily-phet** [comment]
@zepumph re 

> We want to "play with silence," and to remove the continuous sound when the ratio "fitness" is equal to 0. This means we only get sound when you are "getting close" to the ratio. Tagging @emily-phet to see if she disagrees.

The plan sounds great to me!

**Comment by Ashton-Morris** [comment]
[@zepumph I have the files for you here](https://drive.google.com/open?id=10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg)

choir-ahhh-loop.wav
pizz-C4.mp3
pizz-C3.mp3

**Comment by emily-phet** [comment]
To discuss on Tuesday:
- Adjusting rate of staccato (current version sounds a bit frantic) @zepumph can you make sure it's straightforward to change/tune the tempo of the marimba/pizz sound?
- Consider reversing the mapping, where tempo decreases markers approach success. This require sounds to be playing whenever the markers are in use (which we decided last week we didn't want) but I think this is useful to consider even if just conceptually, especially if we're become less confident that the current mapping (tempo increases as markers approach success) will be effective without the sound seeming frantic as success is reached. 
- Consider what the sound is when you reach ratio - what will feel satisfying, perhaps calm, and confirm success. Could this be a single sound, different pitch than sounds heard on approach, and then (if not moving markers continuously) silence? 
- General question - what is the smallest tempo change that is readily recognizable?

**Comment by zepumph** [comment]
I added pizz staccato options and the choir ahh velocity option. See master or https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-dev.21/phet/ratio-and-proportion_en_phet.html. Sorry to cut it short to our meeting time!

**Comment by zepumph** [comment]
Nodes from today's design meeting

* Behavior of when the velocity sound plays. Right now it can occur when any staccato sound plays (any fitness >0), even when the tremolo to the third isn't going. @jbphet mentioned it may be nicer to layer on the velocity sound only when you maintain the tremolo sound while moving. If that is the case we need to greatly relax  the threshold for the "tremolo to the third" sound. Right now it has to be pretty much a "perfect" ratio.

* Can we reduce the frequency of the tremolo (right now it is every 40ms).
* @emily-phet asked what can we do to simplify these cues. Less "noise". Increasing frequency until at the right ratio, then a "release" with less sound 
* Explore a single tone at the end, perhaps with some effects/envelope to make it longer (but not continuous). This keeps playing, but is shaped differently than the unsuccessful sounds.
* Single, new, "success" sound. Like a gong. Only once until to leave success and come back.
  * Could be a higher "ding" like a cash register to indicate success.
  * A long, reverby gong (we won't do this).
  * A chord or lightly arpeggiated group a notes.

Action steps:

- [ ] we can get rid of the following sound options:
  * Random clicks
  * Vibrato and strings
  * Single Pitch change
- [ ] add a "No Velocity sound" option
- [ ] @Ashton-Morris to get new pizz sounds with slightly less reverb.
- [ ] New sound options for staccato success:
   * tremolo thirds (current)
   * single note (higher and longer tail) no tritones allowed
   * with chords, try (1,3,5,7Maj) and maybe we go with (1,3,5,8)
     * single chord 
     * single arpeggiated chord

**Comment by zepumph** [comment]
Clean up above.

**Comment by zepumph** [comment]
Above I added `staccatoMaxFrequency`, so that we can prototype the tremolo speed. I also doubled it from 40ms to 80. Here is an example of it doubled even again:

http://phettest.colorado.edu/ratio-and-proportion/ratio-and-proportion_en.html?ea&brand=phet&staccatoMaxFrequency=160

**Comment by zepumph** [comment]
Above I added chords and arpeggio staccato success sound options. All of these have some build in reverb that @jbphet helped me get set up using `soundManager.reverbLevel`.  I don't know that it will be a long term strategy, but it seems valuable to have to this week's testing. 

I will make a dev version shortly to share this work.

**Comment by zepumph** [comment]
https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-dev.22/phet/ratio-and-proportion_en_phet.html

**Comment by emily-phet** [comment]
Thoughts:

Staccato Success Sound:
- tremolo to third - a bit chaotic, doesn't lead me to think there's a single "special" location
- **single high note** - I liked this one best, but I think the single sound needs to feel richer somehow, and stand out more.
- single major 7th chord - same as below
- single major chord - current variant feels like a little 'bump in the road', doesn't stand out enough
- single arpeggiated chord - too long, I think

Ratio Fitness Sounds:
The pizz sounds sound better to meet at the high rate as you approach success, but I like the individual marimba sounds a bit more. Let's try bringing the max rate for all staccato "Ratio Fitness Sounds" sounds down by 50%, and expand by ~25% the range at which you can hear these sounds. @zepumph let me know if you have any questions on this.

**Comment by zepumph** [comment]
>single arpeggiated chord - too long, I think

I decreased the time of arpeggiation from .25 seconds to .1 second, and I like it a bit more. Actually I think it is my favorite staccato success. I feel like it helps solve the issue you had with the single "chord" strategies too. what do you think?

>single high note - I liked this one best, but I think the single sound needs to feel richer somehow, and stand out more.

Agreed! I wonder if we could get a few options from @Ashton-Morris on "fuller" sounds to test out.

>The pizz sounds sound better to meet at the high rate as you approach success, but I like the individual marimba sounds a bit more.

I think that perhaps we would like the marimba a bit more with less attack, and I think we would like the pizz a bit more with less reverb. Not sure which one I prefer.

>Let's try bringing the max rate for all staccato "Ratio Fitness Sounds" sounds down by 50%

I doubled it from 80ms to 160ms. You can also fine tune this with `?staccatoMaxFrequency=X`.

and expand by ~25% the range at which you can hear these sounds. 

I increased the tolerance from .1 to .125, you can fine tune this with `?tolerance=X`. Also please note that if this tweaking isn't cutting it, we can raise the priority and/or try to finish design on tolerance over in #14.

Please review
.

**Comment by emily-phet** [comment]
@zepumph I tried, using the query parameters listed above, adjusting the tolerance (which worked for me) and the staccatoMaxFrequency (which I couldn't get to work). Can you confirm that this works, and also remind me of any symbols I need to include if using two query parameters (isn't there an ampersand needed or something?). Note, I was trying the query parameters individually just now, so combining two query parameters is not my main issue...

**Comment by Ashton-Morris** [comment]
[Sounds Incoming
](https://drive.google.com/open?id=10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg)

- glock-marimba-c-maj-7-chord-stab.mp3
- glock-marimba-c-maj-7-chord-stab.wav
- glock-marimba-c-maj-chord-stab.mp3
- glock-marimba-c-maj-chord-stab.wav
- glock-marimba-c-maj-arp.mp3
- glock-marimba-c-maj-arp.wav
- glock-marimba-c-maj-7-arp.mp3
- glock-marimba-c-maj-7-arp.wav
- c4-ding-ring-out.mp3
- c4-ding-ring-out.wav
- c4-marimba-long-ring-out.mp3
- c4-marimba-long-ring-out.wav
- marimba-c-maj-7-chord-stab.mp3
- marimba-c-maj-7-chord-stab.wav
- marimba-c-maj-chord-stab.mp3
- marimba-c-maj-chord-stab.wav
- marimba-c-maj-arp  .mp3
- marimba-c-maj-arp  .wav
- marimba-c-maj-7-arp.mp3
- marimba-c-maj-7-arp.wav

**Comment by zepumph** [comment]
@emily-phet I was able to reproduce your problem. I fixed it in master above. Please try the above link again, and note that the first query parameter gets prefixed with a `?`, and each subsequent gets a `&`, so phettest can look like:

http://phettest.colorado.edu/ratio-and-proportion/ratio-and-proportion_en.html?brand=phet&ea&staccatoMaxFrequency=20&tolerance=.3

**Comment by zepumph** [comment]
Updated sounds coming in hot.

**Comment by zepumph** [comment]
There are a lot of sounds in. It would be good to get some feedback on them, as well as the ones in the sim already, before doing the next batch of iteration in the sim. I personally like how some of the sounds have some reverb at the end of the note/chord. I think that will work well if someone stays on the ratio, and then we can fade it out when going out of proportion. I think my favorite is the glock-marimba arpeggio. I think it could fit nicely with the choir velocity sound.

**Comment by zepumph** [comment]
That said, I will exchange out the pizz sounds, as we have already discussed that.

**Comment by emily-phet** [comment]
Here are some thoughts:

**Staccato Max Frequency:** 
- For the marimba sound, I thought 50 is too fast, 100 is ok, 125 is ok (my personal preference when playing with sim), 150 is probably too slow. Let's go with 100 for now. Better to err on the slightly too fast side, I think, if we're between to favorites. With pizz, everything sounded a bit faster than marimba (like an auditory effect, probably not actually faster). We may want to tune the values slightly slower even, if we go with pizz or fizz-like sounds.

**Timbre** 
- I prefer marimba over pizz, but wish it didn't sound so slot-machine like. Any chance @Ashton-Morris you have ideas for a relative of a marimba that has some of the same qualities, but is less slot machine-y when heard in repetition? For pizz - I like the pizz C3 better than pizz C4 now (@zepumph I think you called it last week :)). 

**For the sounds @Ashton-Morris added,** 
- I like the ding-ring-out for the success sound best (outside the sim). We know sounds can be tricky and not turn out how you think in the sim though...
-  I liked listening to the arpeggios, but think the short ding-ring-out, or the *-stab sounds will work better in context. It would be good to cue people they overshot by making sure the special success sound is heard even if you zoom by it moving fast, and I think that will be harder to convey/interpret with an arpeggio. 

@zepumph - Can you add the ding-ring-out as an option to the success sound list (feel free to remove any you think aren't being useful anymore from that list). 

**Subtle difference in sound when above/below success point.** 
- I think we should discuss this today. I thought of another reason this would be helpful, beyond what @jbphet has brought up before...if you're using the keyboard you're moving in 1-gridline increments by default. You could overshoot the success point and still hear the same staccato sound rate - essentially meaning two sequential keyboard presses give you the same sound feedback even if you've moved from just above/below to just below/above the success point. Would be good to have above/below be slightly different. 

**Balancing pickup/drop sounds** 
- For a little bit later in the sound design process...we should probably have the pickup/drop sounds be more in the background when you're also going to hear staccato ratio feedback sounds, but foregrounded more when you won't hear staccato ratio feedback sounds. 

I also have some additional thoughts on tolerance (I think we should try having a 4-gridline total range of feedback across the board). I'll put these comments in #14.

**Comment by jbphet** [comment]
Thoughts:

- The sound design is really coming along!
- The choir sound, IMO, is awesome.
- The current "Single Arpeggiated Chord" option is good, but too quick in my opinion.  I'd prefer the glock-marimba-c-maj-arp from above, though a little bit faster than it is and with less of a tail.
- The major and major 7th chords (not arpeggios) also feel too quick to me.
- I feel like the "success" sound should play every time the user moves through the correct proportion, even if they do so quickly.  This will make them think, "Oh, there's something there that matters."  This will be especially important for non-visual users.
- All of the sounds fade out entirely when not used, which gives the sound design a bit of an unnatural feel.  IMO, the sound generators should be designed such that their sounds are always allowed to finish once initiated, but they stop initiating them when the aren't needed.
- When moving the markers with the keyboard, this cutting off of the sounds is particularly noticeable when moving quickly.
- The tremolo success sound was hard to get on the current master version, not sure why.  I don't think it matters that much though, since most of the other success sounds seem superior.
- It's possible when using the keyboard to get the markers set up in such a way that it's hard to get to "full success", especially if using some mouse combined with some keyboard (I can demo at the meeting).  Should we consider having the (non-shift) arrow keys move to the next tick mark instead of always a full increment?
- When playing with this on a touch screen, I found the sounds generated while moving both markers to be a bit overwhelming, since all of them tended to sound at various times.  I think we should have a bit more hysteresis for the in/out behavior, and once the user is moving and keeping things green, they should get the choir and that's all.
- Is there something being designed for the visual portion to match the "super-success" state where the choir sings (i.e. where the user is moving both markers and maintaining the proportion)?  If not, might I suggest something like a glowing yellow border?
- I'm not sure if this is an issue or not, but I noticed it while testing and thought we should make sure we want the behavior this way.  Due, I suspect, to the nature of ratios, the feedback about whether being close to the correct ratio is asymmetrical on the linear grid that we present.  For example, if "Challenge 1" is selected (1:2) and the left marker is at 2, sound will be produced if the right marker is at 5 (1 too many), but not at 3 (1 too few).

**Comment by zepumph** [comment]
Decisions from today's meeting:

- [ ] We prefer the marimba sound to the pizz sounds. That said it leans a bit towards a "casino sound." @Ashton-Morris will work on some fine tuning to the marimba sound so that it is more like a "bing" and less like a "ding."
- [ ] As for the frequency of the staccato notes (which @jbphet pointed out is actually a period of time between sounds), we will update the default to be `120` and change the name to reflect "period" instead of "frequency."
- [ ] moving proportion sound tolerance (perhaps half of the tolerance)
- [ ] Fine tuning step to make sure that you could change directions of velocities without stopping the choir sound.
- [ ] "jarring ending of sounds" audit needed (superset of the above bullet)
- [ ] fading in/out of moving in proportion sound.
- [ ] @jbphet mentioned that you can move through and past a ratio success sound without hearing the success. Perhaps it would be nice to always have a success sound when there has been a success, even though the ratio success didn't linger. Perhaps the important thing is just to make sure that "some sound" plays in this case. We want to discuss this further in a future design meeting.
- [ ] For staccato ratio success sounds, we would like to hear c4-ding-ring-out.mp3 and glock-marimba-c-maj-7-arp.wav in the sim. These both have long tails, and it would potentially be nice to play the tail for as long as the ratio stays in proportion (sorta like a continuous-feel), but then to dampen it when moving out of proportion so that it doesn't linger too long. (implementation notes: use setOutputLevel's second arg, `timeConstant`, to fade that more).
- [ ] We feel good about removing all of the current staccato success sounds, replacing them for the above checkbox.
- [ ] Move SoundClipChord.js into tambo

**Comment by zepumph** [comment]
>@jbphet mentioned that you can move through and past a ratio success sound without hearing the success. Perhaps it would be nice to always have a success sound when there has been a success, even though the ratio success didn't linger. Perhaps the important thing is just to make sure that "some sound" plays in this case. We want to discuss this further in a future design meeting.

I reworked the logic for the staccato success/in proportion sound, and I think this is much better. Before the algorithm only tried to play the success sound based on the same loop delay as for the normal marimba approaching sound. This meant that the success sound was not eligible for playing every `step`. Which it should be, so I pulled the success sound out of the logic that sees if enough time has passed to warrant a new approach sound. If that doesn't make sense, then try it out and see if it is better. I'm going to check this box off, but please uncheck if there is more to discuss.

**Comment by Ashton-Morris** [comment]
[Here are some Marimba variations to try out](https://drive.google.com/open?id=10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg)

- marimba-variations-003.mp3
- marimba-variations-003.wav
- marimba-variations-002.mp3
- marimba-variations-002.wav
- marimba-variations-001.mp3
- marimba-variations-001.wav
- marimba-variations-004.mp3
- marimba-variations-004.wav
- marimba-variations-000.mp3
- marimba-variations-000.wav

**Comment by zepumph** [comment]
From today's sound design meeting:
- [ ] We want to see the above sounds as options in the sim. We don't need to see option (1) or (4)
- [ ] We want to try changing the pitch based on fitness. Perhaps trying a pentatonic scale, or maybe whole tone.
- [ ] @Ashton-Morris will experiment with some non-tonal sounds (since we ditched the original tremolo-to-third approach).
- [ ] @jbphet mentioned that it is potentially possible for web audio to give you a notification when `setOutputLevel` is finished with its time-delayed level mutation. I may want to try to use that for the staccato success sound. I will create an issue in tambo to investigate if it is needed.
- [ ] If I have way too much time to spend, then I could try changing the c major sin sound to layering in chords instead of volume, based on fitness.

**Comment by zepumph** [comment]
Added the variations on marimba above. Note I also made the staccato sound quieter (along with the success sound).

**Comment by emily-phet** [comment]
@zepumph I'd love to hear the staccato sounds with the pitch change, but at the moment (without that in place), i like the last option (pluck against drywall) the best. It has just enough of a natural feel to not sound like it could be an error indicator (which the ploink and thoink have a little of).

**Comment by Ashton-Morris** [comment]
[Here are some non pitched sounds to try. ](https://drive.google.com/drive/u/1/folders/10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg)They are more like button or UI sounds, some of which are from our previous UI tests.

non-melodic-indicator-1.wav
non-melodic-indicator-1-002.mp3
non-melodic-indicator-1-002.wav
non-melodic-indicator-1-001.mp3
non-melodic-indicator-1-001.wav
non-melodic-indicator-1.mp3

**Comment by zepumph** [comment]
From sound design meeting today:

- [ ] implement the variation in TODOs in StaccatoFrequencySoundGenerator
- [ ] Include a way to have marimba tone with frequency modulation but NO pitch modulation, for interview comparisons
- [x] Discuss with @jbphet about how to handle added code to MultiClip.js

**Comment by zepumph** [comment]
Alright, I think that I got to everything we discussed today during sound design meeting. Here is a version with the "random mix" of sounds for staccato by default, and also an option to use the marimba. When on the marimba sound, the pitch won't modulate.

https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-dev.27/phet/ratio-and-proportion_en_phet.html

**Comment by zepumph** [comment]
Next design meeting it would be nice to take a quick look at the layering done in the c major sin waves. I doubt we will change our strategy now, but it sounds pretty nice no matter.

**Comment by zepumph** [comment]
We also never discussed @Ashton-Morris's non-melodic sounds. Otherwise, I have done all for this issue that I can right now. Unassigning.

**Comment by emily-phet** [comment]
To discuss:

- [ ] Ashton's non-melodic sounds.
- [ ] @zepumph layered C major sin waves (if not for this sim, would be interesting to consider mentioning this for Waves Intro graph challenge.
- [ ] A version of the sim without tempo being used - so just movement results in single sound feedback. @zepumph - can you make a version where there is no repetition of sounds as you're interacting? I like the random mix, but the repetition I think makes it harder to interpret what's happening. I think reviewing in advance some clips from the latest interviews (seeing how others interact, how much of their movement is faster vs slower, etc.) will help us interpret if this is an issue or not.
- [ ] Simple success sounds. Right now I like the "Random Mix" with "Single High Note", but if this is the general path forward would need some tweaking to both sets of sounds.
- [ ] Probably not enough time or final decisions on staccato sound and success sound... but will eventually need to revisit the simultaneous movement success sound.

**Comment by zepumph** [comment]
Discussion from design meeting today:

* It would be good to make some decisions, to help us focus on a prototype path
* Perhaps we can experiment with a DMZ of sorts, in which the staccato success not only has to leave its range, but also pass through a DMZ threshold surrounding it in order to then be able to trigger the staccato success again.

**Comment by zepumph** [comment]
I added a time-consistent staccato sound option above, and made it default. There are two query parameters to control it. The first has a use in the frequency modulated version also (staccatoMinRepeatTime), and in this mode it is the min time between to sounds, regardless of if the fitness changed enough to trigger that sound. The second is what the fitness change threshold is (fitnessChangeThreshold).

Here is an example of a very ugly version of these query parameters, they play to often back to back, and with too little fitness change: http://phettest.colorado.edu/ratio-and-proportion/ratio-and-proportion_en.html?ea&brand=phet&staccatoMinRepeatTime=10&fitnessChangeThreshold=.05

The defaults are http://phettest.colorado.edu/ratio-and-proportion/ratio-and-proportion_en.html?ea&brand=phet&staccatoMinRepeatTime=120&fitnessChangeThreshold=.2 (same as omitting the query parameters)

Please see https://github.com/phetsims/ratio-and-proportion/blob/6cd3eb6c6840c1571244327aa0765e76b1b464b9/js/common/RatioAndProportionQueryParameters.js#L27-L44 for more explanation on the query parameters.

- [ ] Next, hopefully later this afternoon, I will work on a DMZ pattern for the success sound, and maybe pull that into its own sound generator (since it isn't really a staccato feature).

**Comment by Ashton-Morris** [comment]
[Here are some new success sounds.](https://drive.google.com/drive/u/1/folders/10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg)

- sucess-indicator-elec-piano-2w.mp3
- sucess-indicator-elec-piano-2w.wav
- sucess-indicator-elec-piano.mp3
- sucess-indicator-elec-piano.wav
- sucess-indicator-eastern-plucked-instrument-2.mp3
- sucess-indicator-eastern-plucked-instrument-2.wav
- sucess-indicator-eastern-plucked-instrument.mp3
- sucess-indicator-eastern-plucked-instrument.wav

**Comment by zepumph** [comment]
From today's sound meeting:

- [ ] Remove the sound mode added last week that didn't increase frequency as you go closer to in Proportion.
- [ ] Discretize the tones to semitones.
- [ ] ~Investigate https://github.com/phetsims/tambo/issues/114 as an alternative to needing many different notes of the same sound from @Ashton-Morris to get each to have the same length (since we modulate playback rate to change pitch).~ We decided there was no time right now for this. Instead @Ashton-Morris will eventually make sounds for each tone and sound clip combo.

**Comment by zepumph** [comment]
Now I am just waiting on sounds from @Ashton-Morris. Unassigning.

**Comment by Ashton-Morris** [comment]
All right @zepumph [here are these files ](https://drive.google.com/drive/u/1/folders/10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg)

- marimba-variation-v2-c.mp3
- marimba-variation-v2-c-001.mp3
- marimba-variation-v2-c-002.mp3
- marimba-variation-v2-c-sharp-001.mp3
- marimba-variation-v2-c-sharp-002.mp3
- marimba-variation-v2-c-sharp.mp3
- marimba-variation-v2-d.mp3
- marimba-variation-v2-d-001.mp3
- marimba-variation-v2-d-002.mp3
- marimba-variation-v2-d-sharp-001.mp3
- marimba-variation-v2-d-sharp-002.mp3
- marimba-variation-v2-d-sharp.mp3
- marimba-variation-v2-e-001.mp3
- marimba-variation-v2-e-002.mp3
- marimba-variation-v2-e.mp3
- marimba-variation-v2-f-002.mp3
- marimba-variation-v2-f.mp3
- marimba-variation-v2-f-001.mp3
- marimba-variation-v2-f-sharp-001.mp3
- marimba-variation-v2-f-sharp-002.mp3
- marimba-variation-v2-f-sharp.mp3
- marimba-variation-v2-g-001.mp3
- marimba-variation-v2-g-002.mp3
- marimba-variation-v2-g.mp3

**Comment by Ashton-Morris** [comment]
My own personal notes have this "Create two versions of the C G C G organ sound and send that in" I think I was going to create a second success variation so it wouldn't sound too repetitive.

@zepumph The dev build wasn't published, can you let me know which organ sound we had in use so I can create a variation of that?

**Comment by zepumph** [comment]
From design meeting today:
- [ ] We want staccato sounds always playing when you are holding a hand with the mouse. For keyboard, we think that perhaps you have to not just focus, but then move once for this behavior to then be constant until next blur.
- [ ] We decided to go with the staccato sound for out of proportion feedback, and to use the "random mix" as the sound for it. All else can be cleaned up!!!
- [ ] Add hysteresis to the success sound, so that it doesn't play too often when going in and out to of Proportion quickly.
- [ ] @Ashton-Morris to create two versions of a success sound modeled off of success-indicator-elec-piano-2w.mp3 from https://github.com/phetsims/ratio-and-proportion/issues/9#issuecomment-651785405. We want the sound slightly shorter in attack (even though it will still ring out), and we want at least two versions so that we don't have the repetition of the same sound over and over (the JT principle). 
- [ ] Change the default to the "ding" success sound so that we can see how we feel about that this week. This is in spite of the fact that we likely will choose an arpeggiated sound due to its ability to "stand out" better.
- [ ] Add "slider bump" sound, also used as the ratio hand passing a grid node. Make this the sound for NumberSpinner. We will start by adding it to the common code. I'll create a separate issue for that.

**Comment by zepumph** [comment]
To discuss for next meeting from me:

* I added hysteresis for the in proportion succes sound, and it can be adjusted with `?hysteresisThreshold`. The default value is .1, which means 10% of the total fitness range above and below the in proportion range. 
* I had a question about the random mix staccato sounds. Would it be ok if they could repeat? There is code that makes sure of no repetition, but I wasn't sure if that was necessary. 

I may add more in later.

**Comment by zepumph** [comment]
I have implemented all my stuff for this issue, and will continue to work on NumberPicker sounds over in https://github.com/phetsims/scenery-phet/issues/612. Unassigning until Tuesday.

**Comment by Ashton-Morris** [comment]
[Here are some faster electric piano sounds with a faster attack](https://drive.google.com/drive/folders/10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg?usp=sharing)

sucess-indicator-elec-piano-2w-faster-attack.wav
sucess-indicator-elec-piano-2w-faster-attack-001.mp3
sucess-indicator-elec-piano-2w-faster-attack-001.wav
sucess-indicator-elec-piano-2w-faster-attack.mp3

**Comment by Ashton-Morris** [comment]
[I tried to create a woop like John suggested. ](https://drive.google.com/drive/folders/1YHLFqMsM_lXnnTT_Ec-yGK86mTqgX9Ec?usp=sharing)

- accordian-box-close.mp3
- accordian-box-close.wav
- accordian-box-open.mp3
- accordian-box-open.wav

I also tried sounds that were closer to a real Accordion but I didn't bounce them because I feel they were too extreme sounding for PhET.

This approach was not appreciated the last time we were creating sounds for opening and closing but [I added some white noise sweeps just in case.](https://drive.google.com/drive/folders/1YHLFqMsM_lXnnTT_Ec-yGK86mTqgX9Ec?usp=sharing) This is a common UI sounds used in applications and I thought it was a good reference point

- accordian-box-close-white-noise2.mp3
- accordian-box-close-white-noise2.wav
- accordian-box-open-white-noise.mp3
- accordian-box-open-white-noise.wav

All in all I think we'll need to iterate on this to get closer to the correct sound.

**Comment by jbphet** [comment]
Issue for adding sound to PhET button: https://github.com/phetsims/joist/issues/637

**Comment by zepumph** [comment]
From design meeting today:

"In Proportion" sound:
* New sound option that only uses roots and fifths (perhaps starting at g4 since that is where the staccato high note is).
* new option that is a major chord (perhaps starting a third above c4 to not start on the same note as the staccato sound).
* Make at least three options.
* Try making this higher than the sounds from this week (in https://github.com/phetsims/ratio-and-proportion/issues/9#issuecomment-658288530). We do not want it as high as the glock overtones from `glock-marimba-c-maj-7-arp`
* Still make these ring out, we will fade them out with WebAudio when moving out of Proportion.

"Moving in Proportion" sound:
- New options should be like choir but more synthetic (less human but not "jump" by van halen)
- Create at least 2 new options to explore this.
- Start at g4 (above the starting in Proportion sound)


We want these options created and delivered by Monday so that I will have time to input them before the next meeting which will be either next Tuesday or Wednesday.

**Comment by zepumph** [comment]
I have nothing to do in this issue until there are sounds for me to add to the sim. Unassigning.

**Comment by Ashton-Morris** [comment]
"We want these options created and delivered by Monday so that I will have time to input them before the next meeting which will be either next Tuesday or Wednesday." [ Just read this again. Hope you get time!
](https://drive.google.com/drive/folders/10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg?usp=sharing)
**"In Proportion" sound:**
in-proportion-fifths-option-1.wav
in-proportion-major-chord-option-3.mp3
in-proportion-major-chord-option-3.wav
in-proportion-major-chord-option-2.mp3
in-proportion-major-chord-option-2.wav
in-proportion-major-chord-option-1.mp3
in-proportion-major-chord-option-1.wav
in-proportion-fifths-option-3.mp3
in-proportion-fifths-option-3.wav
in-proportion-fifths-option-2.mp3
in-proportion-fifths-option-2.wav
in-proportion-fifths-option-1.mp3

**"Moving in Proportion" sound:**
moving-in-proportion-loop-option-4.wav
moving-in-proportion-loop-option-3 .wav
moving-in-proportion-loop-option-2.wav
moving-in-proportion-loop-option-1.wav

**Comment by zepumph** [comment]
All options above have been implemented. They can be found in this dev link. Looking forward to discussing them tomorrow!

https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-dev.38/phet/ratio-and-proportion_en_phet.html

**Comment by zepumph** [comment]
Discussion about the Moving in Proportion options: 
- [ ] Make the moving in Proportion sound mutually exclusive with the staccato sounds, implementation wise this means making movingInProportionThreshold and MOVING_IN_PROPORTION_FITNESS_THRESHOLD add up to 1.  Already implemented in https://github.com/phetsims/ratio-and-proportion/commit/77f0c610f1c41928c87f166460a1ec64da24fd73, but should be cleaned up to just use a single value.
- [ ] Create a sound option that includes both the "peaceful tranquility" and the "choir sound"
- [ ] Only keep four options, the aforementioned mix, choir, and "peaceful tranquility," and "dream time"
- [ ] @Ashton-Morris to create a sound that layers the choir sound plus a rendition of "peaceful tranquility" (moving-in-proportion-loop-option-4.wav), and removes a few of the lower bass notes so that it floats above the in proportion sound. (less breadth to the sound).


Discussion about the in Proportion options:

- [ ] The hysteresis option is weird when just moving a single value, as fine tuning it can yield just a flash of green, and no sound. Perhaps it should only be implemented for when moving in proportion. 
- [ ] Get rid of both "option 3"



Thoughts about how the moving in proportion sound relates to the in proportion
* Maybe the in proportion sound can be swapped out for the moving in proportion sound, instead of layering on top of.
* Perhaps we can just make the in proportion sound quieter when the moving in proportion sound is playing
* Maybe we can make the moving in proportion sound actually include themes or elements of the in proportion sound in it (and to just play that). Note the complexity of adding in a single ringing "ding" into a looping sound.

**Comment by emily-phet** [comment]
@zepumph @BLFiedler - I played with the sim this morning quite a bit. I think the dev 40 paradigm for mutually exclusive in  proportion, success, and choir sounds is good. 

I think the choir sound could tail off faster. Sometimes it feels like I can still hear it even if I'm quite far off the ratio position.

If @BLFiedler agrees, I think we move to all using the sim with the different sound combinations, and listing out preferred options (considering all the sounds together).

**Comment by BLFiedler** [comment]
Yeah - I think this is plenty smooth. The staccato and in-prop sounds seem to play when you move far enough out and/or re-enter just fine. 

re: choir sound tailing off: I think this is particularly noticeable when you lock the ratio. There is a time delay between ending movement and the choir sound stopping. I agree that delay could be shorter so it's not still playing if you stop or leave the proper proportion.

Side note: @zepumph discovered If you select the Peaceful Tranquility velocity sound and then hit the reset button (iPad, Safari) it locks all interaction with the simulation. Only that sound option though. Made an issue: #97

**Comment by zepumph** [comment]
>Side note: @zepumph, I discovered if you select the Peaceful Tranquility velocity sound and then hit the reset button (iPad, Safari) it locks all interaction with the simulation. Only that sound option though. Made an issue: #97

Should now be fixed

>I think the choir sound could tail off faster. Sometimes it feels like I can still hear it even if I'm quite far off the ratio position.

This could be a bug, or could be our current fade out time of .2 seconds. I'll try decreasing that to .1 to see if that is better. I'll get back to you after some investigation.

All other sound options have been implemented on master.

**Comment by emily-phet** [comment]
Sounds like we're all set to come up with our list of favorite combinations, in preparation for Tuesday. I'll let all know to do this in the RaP slack channel. :)

**Comment by emily-phet** [comment]
Note - there are still some remaining to dos for @zepumph and @Ashton-Morris in this: https://github.com/phetsims/ratio-and-proportion/issues/9#issuecomment-662725606 above.

**Comment by Ashton-Morris** [comment]
[I have this here. ^__^](https://drive.google.com/drive/folders/10-kHZgFSu0HEMUfNk0zEn0cvPRK8gFXg?usp=sharing)

- moving-in-porportion-loop-preaceful-and-choir.wav
- moving-in-porportion-loop-preaceful-and-choir.mp3

**Comment by emily-phet** [comment]
My thoughts re staccato success sound and velocity sound.

I'd be fine with any combination of "Fifths option 2", "Chord option 2" for the staccato success sound, and "Choir" or "Peaceful + Choir = nirvana" :) sound. 

My top two favorites are:
1) Fifths option 2 + choir
2) Chord option 2 + choir

Some thoughts on these preferences: 

- I like the fifths option 2 slightly more than chord option 2 for this purpose because fifths sounds a little more "smooth" and chord rings out really bright and loud (in a good way, but it does stand out a lot). 

- I like choir slightly more than choir+ because it's a bit simpler, and doesn't feel as "full" as the choir+ sound, which (to me) may sound a little odd if you're trying to use the feedback in a group setting turned down low. Since it's so "full" sounding, it feels to me like it's asking to be listened to louder and as more of the focus for some reason.

**Comment by emily-phet** [comment]
@zepumph There were times when I thought I was hearing sound feedback when the screen looked white. Is this the case, or is the tolerance such that the screen just happens to look white because it's a super subtle shade of green? 
If it's the case that the background is exactly white but providing sound feedback, we should discuss if we want this to happen tomorrow.

**Comment by zepumph** [comment]
After https://github.com/phetsims/ratio-and-proportion/commit/3523bc99cd94018890f94f2c585e5d4a2a4526f9, now the `?movingInProportionThreshold` query parameter takes the value in "distance from max fitness" instead of distance from min fitness. So the value used to be .7, but now it is .3.

**Comment by zepumph** [comment]
Notes from adding the new combined `wav` sound:
* I added the new wav file from @Ashton-Morris and named it `moving-in-proportion-loop-peaceful-and-choir.wav`.
* I can't actually hear a difference between the two sounds I combined in code, and @Ashton-Morris's. Did you take out some of the bass notes as described in https://github.com/phetsims/ratio-and-proportion/issues/9#issuecomment-662725606? I think my ears may be failing me.

**Comment by zepumph** [comment]
Everything has been implemented from the above checkboxes.

>If it's the case that the background is exactly white but providing sound feedback, we should discuss if we want this to happen tomorrow.

I have not seen this, but will be on the lookout. Please create a new issue if you can consistently reproduce. I'm pretty sure that it is just a very faint shade of green. Hopefully logging added in #103 will help see that.

---------------------------

My thoughts about combos:
* I agree with @emily-phet that option 2 is a bit better than option 1, and I like fifths more than chords (yay fifths 2)!
* I like the choir+ sound most.
* Thus my vote is fifths2 + Choir+ peace
* I think that fifths 2 could use some minor tweaking to have a bit less twang at the end. It is interesting on its own, but it is a bit muddy when combined with the "moving in proportion" sound. I feel this may no matter what moving in proportion sound is selected.

Looking forward to tomorrows chat!

**Comment by jbphet** [comment]
As requested by @emily-phet, I spent some time reviewing the latest dev version (https://phet-dev.colorado.edu/html/ratio-and-proportion/1.0.0-dev.41/phet/ratio-and-proportion_en_phet.html) and have the following comments:

- The hysteresis for playing the "in proportion" sound is too much when not doing dual drag.  I know I advocated for having some hysteresis, so I should elaborate.  When using the sim on a touchscreen and moving both markers, the "in proportion" sound played and stopped a lot, and this seemed distracting, so hysteresis seemed important.  However, when using v1.0.0-dev.41 with a mouse, it feels odd that the sound isn't played each time the optimal proportion is reached and the max green is shown.  So, perhaps the hysteresis should only be used when both markers are being dragged, or maybe the amount should be much less when one marker is dragging versus both markers.  This problem was particularly noticeable when using fine-grained keyboard-based movement (shift-arrow).
- The "in proportion" sound should play every time the threshold is crossed.  As it is now, some fast crossings don't cause it to play.
- My favorite combination of sounds is also "Choir/peaceful tranquility blend" and "Fifths option2".  The former seems like the clear winner to me for the "Velocity Sound".  For the latter (currently labeled "Staccato Success Sound"), I would also be okay with "Arpeggiated maj7 chord" or "Single ding high note".

**Comment by zepumph** [comment]
>The hysteresis for playing the "in proportion" sound is too much when not doing dual drag.
and
>The "in proportion" sound should play every time the threshold is crossed. As it is now, some fast crossings don't cause it to play.

This was fixed in master in https://github.com/phetsims/ratio-and-proportion/commit/b643b7a894efc85a2c3900d42cdbaa8516e9c034 and was a bullet from a previous design meeting, see 
https://github.com/phetsims/ratio-and-proportion/issues/9#issuecomment-662725606:

>The hysteresis option is weird when just moving a single value, as fine tuning it can yield just a flash of green, and no sound. Perhaps it should only be implemented for when moving in proportion.

**Comment by BLFiedler** [comment]
The velocity sounds are reaaalllyyy quiet for me relative to the in/out of prop sounds, so this was a little harder to assess how rich/crowded the soundspace was, but: 

- **MovingInProp**: Choir/Peaceful Choir are on nearly equal footing for me. I can see how the soundspace could be a little overpacked. I'd be fine with either. I suspect those using external speakers won't really hear the added parts in PeacefulChoir.
- **InProp**: I am fine with either Option 2 or the Arp Maj 7th. I might like the Arp Maj 7th the best if the notes the choir was singing meshed up a bit better? Something about the softness of that success sound is still appealing to me for repeated listens.

Could we maybe consider playing a minimum amount of a success sound that we consider the "core" part before it rings out? It does feel a little weird to let go and have some of the arpeggiated sound cut out so quickly. This may also address John's comment on sounds sometimes not playing. I suspect it may be triggering, but getting cut off just as quickly?

**Comment by zepumph** [comment]
- [ ] Moving in proportion sound is much quieter than the in proportion sound. We should balance those by probably meeting somewhere in the middle. 

- [ ] We decided as a group on Choir + Peaceful and Fifths 2. It may need some fine tuning, but we want to go through a round of interviews before doing more work on this (in case we need to pivot).

**Comment by zepumph** [comment]
Some notes to look at when (if) we come back to fine tune the `Choir + Peaceful and Fifths 2` sound combo:
* For now we are keeping the bassy breadth of the peacful tranquility sound in because it sounds "complete," but in the future, we may change how we feel to thinking it a bit too muddy. We will come back to this thought
* The balance of levels between Moving in proportion sound vs in proportion will likely need to be tweaked beyond what is done for https://github.com/phetsims/ratio-and-proportion/issues/9#issuecomment-665287475.

**Comment by zepumph** [comment]
- [ ] When you would hear the staccato sound, don't play pick up and drop UI sounds for the hands.

**Comment by zepumph** [comment]
Once the above is implemented, we can close this issue. As stuff comes out of interviews, we can make more issues for it.

**Comment by zepumph** [comment]
Alright! All has been implemented and cleaned up. Let's create new issues from here as we fine tune our sounds. Good work team!
